components/tabs/src/schema.rs

```diff
@@ -67,4 +102,47 @@ mod tests {
         conn.execute_batch(CREATE_SCHEMA_SQL)
             .expect("should allow running twice");
     }
+
+    #[test]
+    fn test_tabs_meta() {
+        let mut db = TabsStorage::new_with_mem_path("test");
+        //let conn = db.open_or_create().unwrap();
+        let test_key = "TEST KEY A";
+        let test_value = "TEST VALUE A";
+        let test_key2 = "TEST KEY B";
+        let test_value2 = "TEST VALUE B";
+
+        db.put_meta(test_key, &test_value).unwrap();
+        db.put_meta(test_key2, &test_value2).unwrap();
+
+        let retrieved_value: String = db.get_meta(test_key).unwrap().expect("test value");
+        let retrieved_value2: String = db.get_meta(test_key2).unwrap().expect("test value 2");
+
+        assert_eq!(retrieved_value, test_value);
+        assert_eq!(retrieved_value2, test_value2);
+
+        // check that the value of an existing key can be updated
+        let test_value3 = "TEST VALUE C";
+        db.put_meta(test_key, &test_value3).unwrap();
+
+        let retrieved_value3: String = db.get_meta(test_key).unwrap().expect("test value 3");
+
+        assert_eq!(retrieved_value3, test_value3);
+
+        // check that a deleted key is not retrieved
+        db.delete_meta(test_key).unwrap();
+        let retrieved_value4: Option<String> = db.get_meta(test_key).unwrap();
+        assert!(retrieved_value4.is_none());
+    }
+
+    #[test]
+    fn test_tabs_db_upgrade_from_v1() {
+        let db_file = MigratedDatabaseFile::new(TabsMigrationLogic, CREATE_V1_SCHEMA_SQL);
+        db_file.run_all_upgrades();
+        // Verify we can open the DB just fine, since migration is essentially a drop
+        // we don't need to check any data integrity
+        let mut storage = TabsStorage::new(db_file.path);
```

Comment:
this should check the migration actually worked (eg, selecting from the GUID column would fail if it didn't, even with no rows)